### PR TITLE
[MINOR][SQL] Fix error message for `UNEXPECTED_INPUT_TYPE`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to correct the minor syntax on error message for `UNEXPECTED_INPUT_TYPE`,

### Why are the changes needed?

Error message should be started with upper-case character, and clear to read.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

```
./build/sbt “sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*”
```

### CODE REVIEW

**Comment by itholic** [comment]
I made it into a separate PR from my other tasks although it's very small change, to avoid in case this error message affecting the tests on the original PR.

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -399,7 +399,7 @@
       },
       "UNEXPECTED_INPUT_TYPE" : {
         "message" : [
-          "parameter <paramIndex> requires <requiredType> type, however, <inputSql> is of <inputType> type."
+          "Parameter <paramIndex> requires <requiredType> type, however <inputSql> is of type <inputType>."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:402]:
```suggestion
          "Parameter <paramIndex> requires the <requiredType> type, however <inputSql> has the type <inputType>."
```

Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Thanks!

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merged to master.
Thank you, @itholic.

**Pull request closed without merging**